### PR TITLE
Prototype training process

### PR DESCRIPTION
**Pull request by foldfelis**

### CODE REVIEW

File: test/dense.jl [hunk 1]
```diff
@@ -15,3 +15,71 @@ end
...
+        ps = []
+        for layer in layers
+            model = layer ∘ model
+            for param in Scapa.params(layer)
+                push!(ps, param)
+            end
+        end
+
+        return model, ps
+    end
```
Comment (chengchingwen) [inline test/dense.jl]: I think with this design, it would be hard to define function with branches.

File: test/dense.jl [hunk 2]
```diff
@@ -15,3 +15,71 @@ end
...
+
+        # ### update params ###
+
+        i = 1
+        for layer in layers
+            Scapa.update!(layer, ps[i:(i+Scapa.nparams(layer)-1)])
+            i += Scapa.nparams(layer)
+        end
+
+        model, ps = get_model_and_params(layers)
```
Comment (chengchingwen) [inline test/dense.jl]: Why do we need to call this in each iteration?

Comment (foldfelis) [reply test/dense.jl]: It looks like the `\circ` function seem won't update after `ps` is updated.

File: test/dense.jl (hunk 1 above)
Comment (foldfelis) [reply test/dense.jl]: Yes, ur right XD, But let's focus on the implementations of layers

**Comment by foldfelis** [comment]
It is a crappier prototype that needs to be redesign.

**Pull request merged**